fsspec/implementations/prefix.py

```diff
@@ -0,0 +1,163 @@
+from pathlib import Path
+from typing import Any, Iterable, Sequence, Union
+
+import fsspec
+from fsspec import AbstractFileSystem
+from fsspec.core import split_protocol
+from fsspec.spec import AbstractBufferedFile
+from fsspec.utils import stringify_path
+
+
+class PrefixBufferedFile(AbstractBufferedFile):
+    def _fetch_range(self, start, end):
+        pass
+
+
+class PrefixFileSystem(AbstractFileSystem):
+    def __init__(
+        self,
+        prefix: str,
+        filesystem: fsspec.AbstractFileSystem,
+        *args,
+        **storage_options,
+    ) -> None:
+        super().__init__(*args, **storage_options)
+        self.prefix = stringify_path(prefix)
+
+        if not self.prefix:
+            self.prefix = self.sep
+
+        self.filesystem = filesystem
+
+    def _get_relative_path(self, path: str) -> str:
+        if path[: len(self.sep)] == self.sep:
+            return path[len(self.sep) :]
+        return path
+
+    def _add_fs_prefix(self, path: str) -> Union[str, Sequence[str]]:
+        if isinstance(path, (str, Path)):
+            path = stringify_path(path)
+            protocol, path = split_protocol(path)
+
+            path = self._get_relative_path(path)
+
+            if self.prefix == self.sep:
+                path = f"{self.sep}{path}"  # don't add twice the same sep
+            else:
+                path = f"{self.prefix}{self.sep}{path}"
+
+            return protocol + "://" + path if protocol is not None else path
+        elif isinstance(path, Iterable):
+            return [self._add_fs_prefix(x) for x in path]
+        assert False
+
+    def _remove_fs_prefix(self, path: str) -> Union[str, Sequence[str]]:
+        if isinstance(path, (str, Path)):
+            path = stringify_path(path)
+            protocol, path = split_protocol(path)
+            path = path[len(self.prefix) + 1 :]
+            return protocol + "://" + path if protocol is not None else path
+        elif isinstance(path, Iterable):
+            return [self._remove_fs_prefix(x) for x in path]
+        assert False
+
+    def mkdir(self, path: str, create_parents: bool = True, **kwargs) -> None:
+        path = self._add_fs_prefix(path)
+        return self.filesystem.mkdir(path=path, create_parents=create_parents, **kwargs)
+
+    def makedirs(self, path: str, exist_ok: bool = False):
+        path = self._add_fs_prefix(path)
+        return self.filesystem.mkdirs(path=path, exist_ok=exist_ok)
```

Comment:
```suggestion
        return self.filesystem.makedirs(path=path, exist_ok=exist_ok)
```
for symmetry, since we are overriding `makedirs`